drivers/counter/counter_mspm0.c

```diff
@@ -81,22 +78,20 @@ static int counter_mspm0_set_alarm(const struct device *dev, uint8_t chan,
 	}
 
 	struct counter_mspm0_data *data = dev->data;
-	if (k_sem_take(&data->busy_sem, K_FOREVER) == 0) {
-
+	int err = 0;
+	K_SPINLOCK(&data->lock) {
 		struct counter_mspm0_channel_data *channel_data = &data->channel_data[chan];
 		if (channel_data->callback) {
 			LOG_ERR("Callback is already registered");
```

Comment:
Please remove this log. We have the return value - should be fine :+1: 
FYI: `spin_lock` context is critical so only minimal work has to be performed here (as we just discussed). We can not guarantee that the log is offloaded why this might be problematic.